src/Core/Localization/Currency/PatternTransformer.php

```diff
@@ -88,6 +105,55 @@ public function transform(string $currencyPattern, string $transformationType):
         return implode(';', $transformedPatterns);
     }
 
+    /**
+     * Provides currency pattern understandable to symfony, but uses prestashop Locale.
+     * E.g. when passing options to render MoneyType widget
+     *
+     * @param string $localeIsoCode e.g. fr-FR, en-US
+     * @param string $currencyCode e.g. EUR, USD
+     * @param bool $isPatternPositive if false, then "-" is prepended to the pattern
+     *
+     * @return string|null
+     *
+     * @throws InvalidArgumentException
+     * @throws LocalizationException
+     */
+    public function getFrameworkPattern(string $localeIsoCode, string $currencyCode, bool $isPatternPositive): ?string
+    {
+        $priceSpecification = $this->localeRepository->getLocale($localeIsoCode)->getPriceSpecification($currencyCode);
+
+        if (!($priceSpecification instanceof Price)) {
+            throw new InvalidArgumentException(sprintf('Expected instance of %s', Price::class));
+        }
+
+        $patternType = $this->getTransformationType($priceSpecification->getPositivePattern());
+
+        $positivePatternMap = [
+            self::TYPE_LEFT_SYMBOL_WITH_SPACE => sprintf(
+                '%s%s{{ widget }}',
+                self::CURRENCY_SYMBOL,
+                self::NO_BREAK_SPACE
+            ),
+            self::TYPE_RIGHT_SYMBOL_WITH_SPACE => sprintf(
+                '{{ widget }}%s%s',
+                self::NO_BREAK_SPACE,
+                self::CURRENCY_SYMBOL
+            ),
+            self::TYPE_LEFT_SYMBOL_WITHOUT_SPACE => sprintf('%s{{ widget }}', self::CURRENCY_SYMBOL),
+            self::TYPE_RIGHT_SYMBOL_WITHOUT_SPACE => sprintf('{{ widget }}%s', self::CURRENCY_SYMBOL),
+        ];
+
+        if (empty($positivePatternMap[$patternType])) {
+            return null;
+        }
+
+        return str_replace(
+            self::CURRENCY_SYMBOL,
+            $priceSpecification->getCurrencySymbol(),
+            $isPatternPositive ? $positivePatternMap[$patternType] : sprintf('-%s', $positivePatternMap[$patternType])
```

Comment:
If we really want to use the whole CLDR feature I'm not sure we should construct the negative form this way. In most western languages this is how it would be built (although some would have a space) but it's not the case for all languages (including RTL languages) so iI'm not sure we should use this but instead `getNegativePattern()`

And actually I'm thinking even the way you build the pattern may already be breaking the format for some languages. Ideally we should use the returned pattern from the specification and replace the part that we need with `{{ widget }}`

I think you can simply replace `#,##0.00` because it's the common pattern used in every language for the CLDR norm and parser. Maybe it will make this a bit harder to unit test, but it's not that much a problem because it would be more valuable to test this in integration tests, this way it's really based on the CLDR fixtures (if you can keep the unit tests though it's still a good thing).